### PR TITLE
Use `m2-git` on Windows

### PR DESCRIPTION
**Pull request by jakirkham**
The `git` package on Windows is binary repackaged from upstream. However it includes some MSYS2 bits in itself ( https://github.com/conda-forge/git-feedstock/issues/135 ), which clobber other MSYS2 packages we might use. To address this, switch to `m2-git` on Windows.

<hr>

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Review by beckermr — DISMISSED** [review]
This recipe is still noarch but now has arch selectors. Don't we need a git metapackage that selects either m2-git or git for this recipe to depend on?

**Comment by beckermr** [comment]
As it stands now the build makes to artifacts with identical build numbers and different hashes. At minimum, it'd be advisable to put win vs unix in the build string so people can understand the difference.

**Comment by isuruf** [comment]
This recipe follows docs at https://conda-forge.org/docs/maintainer/knowledge_base.html#noarch-packages-with-os-specific-dependencies .

If you think that's wrong, maybe open an issue in website repo?

**Comment by jakirkham** [comment]
@beckermr , please see [this doc]( https://conda-forge.org/docs/maintainer/knowledge_base.html#noarch-packages-with-os-specific-dependencies )

**Comment by beckermr** [comment]
Ohhhhhhh my apologies everyone! Thank you for pointing that out. I still think the build string change would be helpful to others, but otherwise LGTM!

File: recipe/meta.yaml [hunk 1]
```diff
@@ -9,7 +9,7 @@ source:
   sha256: 4a0373e84316cf4bb711779a258a200f778daf3e6cfc51cae8a5df383e3383dd
 
 build:
-  number: 0
+  number: 1
   noarch: python
```
Comment (jakirkham) [inline recipe/meta.yaml:13]: Is this what you had in mind Matt?

```suggestion
  noarch: python
  string: unix_pyh{{ PKG_HASH }}_{{ PKG_BUILDNUM }}  # [unix]
  string: win_pyh{{ PKG_HASH }}_{{ PKG_BUILDNUM }}   # [win]
```

Comment (jakirkham) [reply recipe/meta.yaml:13]: Also just to clarify the hashes are different on their own (taken [from CI]( https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=864825&view=results ))

* UNIX: `conda-smithy-3.30.4-pyh707e725_1.conda`
* Windows: `conda-smithy-3.30.4-pyh7428d3b_1.conda`

Comment (isuruf) [reply recipe/meta.yaml:13]: @jakirkham, can you apply this suggestion? I don't have permissions.

**Comment by jakirkham** [comment]
@conda-forge-admin , please re-render

File: recipe/meta.yaml (hunk 1 above)
Comment (jakirkham) [reply recipe/meta.yaml:13]: Yep added

Was unsure whether we wanted to include this

**Comment by jakirkham** [comment]
@conda-forge-admin , please re-render

**Review by xhochy — APPROVED** [review]

**Comment by h-vetinari** [comment]
FWIW, for me this broke having git understand remote references (showing where HEAD and other remote branches are when doing `git log`).

**Comment by beckermr** [comment]
That is rather confusing to me. Is `m2-git` on windows broken?

**Comment by h-vetinari** [comment]
Possibly. Here's an example of the difference:

Without m2-git (== situation before this PR even with an activated environment):

image

After this PR:

image

My `builder` environment contains smithy; not only are branch heads not shown anymore, but apparently tags are affected too.

**Comment by beckermr** [comment]
Ouch. @jakirkham @isuruf any ideas or comments here?

**Comment by jakirkham** [comment]
Does this have something to do with where `m2-git` looks for `git config` or perhaps the version of `git` used?

**Pull request merged**